edc-client/src/main/java/com/turn/edc/router/RequestRouter.java

```diff
@@ -83,6 +85,29 @@ public boolean store(CacheInstance destination, StoreRequest request) {
 
 		return true;
 	}
+	
+	public boolean setTTL(CacheInstance destination, String key, int ttl) 
+			throws TimeoutException, IOException{
+
+		StorageConnection connection = routingMap.get(destination.hashCode());
+
+		if (connection == null) {
+			logger.info("Existing connection not found, creating new connection: {}", destination);
+			try {
+				connection = connectorFactory.create(
```

Comment:
Could you move this code into a separate method? Specifically the try-catch blocks and the `routingMap` update. Right now this is duplicated code since it's the same logic as in the `get()` method.